src/tests/ftest/erasurecode/restart.py

```diff
@@ -28,46 +58,142 @@ def execution(self, agg_check=None):
 
         Args:
             agg_check: When to check Aggregation status.Either before restarting all the servers or
-                       after.Default is None so not to check aggregation, but wait for 20 seconds
-                       and restart the servers.
+                       after. Default to None.
         """
-        # Write all EC object data to NVMe
-        self.ior_write_dataset(operation="Auto_Write", percent=self.percent)
-        self.log.info(self.pool.pool_percentage_used())
-        # Write all EC object data to SCM
-        self.ior_write_dataset(storage='SCM', operation="Auto_Write", percent=self.percent)
-        self.log.info(self.pool.pool_percentage_used())
-        self.pool.pool_percentage_used()
-
-        if not agg_check:
-            # Set time mode aggregation
+        # 1.
+        ior_w_flags = self.params.get("flags", "/run/ior/iorflags/*")
+        ior_read_flags = self.params.get("read_flags", "/run/ior/iorflags/*")
+        block_transfer_sizes = self.params.get("block_transfer_sizes", "/run/ior/*")[0]
+        obj_class = self.params.get("dfs_oclass_list", "/run/ior/objectclass/*")
+        self.log_step("Create pool and containers")
+        self.pool = self.get_pool()
+        containers = []
+        for oclass in obj_class:
+            # Skip the object type if server count does not meet the minimum EC object
+            # server count
+            if oclass[1] > self.server_count:
+                continue
+            ec_object = get_data_parity_number(self.log, oclass[0])
+            containers.append(
+                self.get_container(
+                    self.pool, daos_command=self.get_daos_command(), oclass=oclass[0],
+                    properties="rd_fac:{}".format(ec_object["parity"])))
+
+        # 2.
+        self.log_step("Disable aggregation")
+        self.pool.set_property("reclaim", "disabled")
+
+        # 3.
+        self.log_step("Get initial pool free space from dmg pool query")
+        initial_free_space = self.pool.get_total_free_space(refresh=True)
+        self.log.info("=Initial pool free space: %s", initial_free_space)
+
+        # 4.
+        self.log_step("Run IOR write all EC object data to container")
+        job_manager = get_job_manager(self, subprocess=None, timeout=120)
+        for oclass, container in zip(obj_class, containers):
```

Comment:
This should be a method that you call 3 times - 1st write, 2nd write, and then read.  Something like `self.run_ior(containers, ior_kwargs)`